### ISSUE TITLE
virt-launcher fails with "Connecting to libvirt daemon failed" on EKS

### ISSUE THREAD

**Issue by seumassoltysik** [kubevirt/kubevirt#11240]
I am running k8s 1.27 on EKS using EC2 instances running AWS Linux optimized for k8s. 
I have emulation enabled:
```
spec:
  certificateRotateStrategy: {}
  configuration:
    developerConfiguration:
      useEmulation: true
```

All Kubevirt pods have initialized successfully.
However when I launch a VM the virt-launcher pod fails with the compute pod throwing the following error:

`{"component":"virt-launcher-monitor","level":"error","msg":"could not create up serial console term file: /var/run/kubevirt-private/215771ad-edd6-4e4b-b4a9-09e5c3e092df/virt-serial0-log-sigTerm","pos":"virt-launcher-monitor.go:72","reason":"open /var/run/kubevirt-private/215771ad-edd6-4e4b-b4a9-09e5c3e092df/virt-serial0-log-sigTerm: no such file or directory","timestamp":"2024-02-13T03:51:17.198585Z"}
{"component":"virt-launcher","level":"info","msg":"Collected all requested hook sidecar sockets","pos":"manager.go:86","timestamp":"2024-02-13T03:51:17.233572Z"}
{"component":"virt-launcher","level":"info","msg":"Sorted all collected sidecar sockets per hook point based on their priority and name: map[]","pos":"manager.go:89","timestamp":"2024-02-13T03:51:17.233625Z"}
**{"component":"virt-launcher","level":"info","msg":"Connecting to libvirt daemon: qemu+unix:///session?socket=/var/run/libvirt/virtqemud-sock","pos":"libvirt.go:563","timestamp":"2024-02-13T03:51:17.233974Z"}
{"component":"virt-launcher","level":"info","msg":"Connecting to libvirt daemon failed: virError(Code=38, Domain=7, Message='Failed to connect socket to '/var/run/libvirt/virtqemud-sock': No such file or directory')","pos":"libvirt.go:571","timestamp":"2024-02-13T03:51:17.234578Z"}**
{"component":"virt-launcher","level":"info","msg":"libvirt version: 9.5.0, package: 6.el9 (builder@centos.org, 2023-08-25-08:53:56, )","subcomponent":"libvirt","thread":"38","timestamp":"2024-02-13T03:51:17.249000Z"}
{"component":"virt-launcher","level":"info","msg":"hostname: testvm","subcomponent":"libvirt","thread":"38","timestamp":"2024-02-13T03:51:17.249000Z"}
{"component":"virt-launcher","level":"error","msg":"Unable to open /dev/kvm: No such file or directory","pos":"virHostCPUGetCPUID:1422","subcomponent":"libvirt","thread":"38","timestamp":"2024-02-13T03:51:17.249000Z"}
{"component":"virt-launcher","level":"error","msg":"internal error: Unable to get session bus connection: Cannot spawn a message bus without a machine-id: Unable to load /var/lib/dbus/machine-id or /etc/machine-id: Failed to open file “/var/lib/dbus/machine-id”: No such file or directory","pos":"virGDBusGetSessionBus:126","subcomponent":"libvirt","thread":"38","timestamp":"2024-02-13T03:51:17.249000Z"}
{"component":"virt-launcher","level":"error","msg":"internal error: Unable to get system bus connection: Could not connect: No such file or directory","pos":"virGDBusGetSystemBus:99","subcomponent":"libvirt","thread":"38","timestamp":"2024-02-13T03:51:17.249000Z"}`

So basically everything looks to be running fine however, when kubevirt tries to create the actual VM, the infrastructure for creating VMs seems to be missing.

I am unclear as to the issue as the requirements docs for deploying Kubevirt are unclear and I am unclear as to whether I am just missing a requirement in my setup.
Here are the questions I have beyond the basic question of why is the launcher failing:

Can I deploy kubevirt on EKS using a VM environment as opposed to bare metal?
If I am using VMs and emulation mode, do I need the libvirt service running on the nodes?
Is libvirt available as a service through one of the pods deployed by kubevirt?
Do I need kubevirt running on the nodes?

**Comment by seumassoltysik** [comment]
One more note, the AWS nodes I am using are the following: https://docs.aws.amazon.com/eks/latest/userguide/eks-optimized-ami.html
The don't appear to have libvirt running or have the kvm linux kernel module installed. 
They also don't seem to support nested virtualization(https://docs.fedoraproject.org/en-US/quick-docs/using-nested-virtualization-in-kvm/index.html) as neither /sys/module/kvm_intel/parameters/nested or /sys/module/kvm_amd/parameters/nested are located on the nodes.

**Comment by dhiller** [comment]
Since you are using emulation, things are pretty much slower than on baremetal machines, which is recommended to use for KubeVirt. We discussed the error, and we think that this might be the initial error, and might eventually go away.

Do you have observed for a longer time? What did happen later on?

**Comment by seumassoltysik** [comment]
Hmm interesting. The VM shows it has been trying to start for the last 48hrs. 
**kubectl describe vm testvm**
shows:    
**Message:               failed to configure vmi network: setup failed, err: open /proc/sys/net/ipv6/conf/all/forwarding: no such file or directory**

Full Output:
`soltyse1@X95JWK6M16 ~ % kubectl describe vm testvm             
Name:         testvm
Namespace:    default
Labels:       <none>
Annotations:  kubevirt.io/latest-observed-api-version: v1
              kubevirt.io/storage-observed-api-version: v1
API Version:  kubevirt.io/v1
Kind:         VirtualMachine
Metadata:
  Creation Timestamp:  2024-02-13T02:23:59Z
  Finalizers:
    kubevirt.io/virtualMachineControllerFinalize
  Generation:  13
  Managed Fields:
    API Version:  kubevirt.io/v1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .:
          f:kubectl.kubernetes.io/last-applied-configuration:
      f:spec:
        .:
        f:template:
          .:
          f:metadata:
            .:
            f:labels:
              .:
              f:kubevirt.io/domain:
              f:kubevirt.io/size:
          f:spec:
            .:
            f:domain:
              .:
              f:devices:
                .:
                f:disks:
                f:interfaces:
              f:resources:
                .:
                f:requests:
                  .:
                  f:memory:
            f:networks:
            f:volumes:
    Manager:      kubectl-client-side-apply
    Operation:    Update
    Time:         2024-02-14T19:39:26Z
    API Version:  kubevirt.io/v1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          f:kubevirt.io/latest-observed-api-version:
          f:kubevirt.io/storage-observed-api-version:
        f:finalizers:
          .:
          v:"kubevirt.io/virtualMachineControllerFinalize":
      f:spec:
        f:running:
    Manager:      Go-http-client
    Operation:    Update
    Time:         2024-02-14T19:44:31Z
    API Version:  kubevirt.io/v1
    Fields Type:  FieldsV1
    fieldsV1:
      f:status:
        .:
        f:conditions:
        f:created:
        f:desiredGeneration:
        f:observedGeneration:
        f:printableStatus:
        f:volumeSnapshotStatuses:
    Manager:         Go-http-client
    Operation:       Update
    Subresource:     status
    Time:            2024-02-14T19:44:37Z
  Resource Version:  256885959
  UID:               d2c77165-b298-4692-b071-a0d321e27b6a
Spec:
  Running:  true
  Template:
    Metadata:
      Creation Timestamp:  <nil>
      Labels:
        kubevirt.io/domain:  testvm
        kubevirt.io/size:    small
    Spec:
      Architecture:  amd64
      Domain:
        Devices:
          Disks:
            Disk:
              Bus:  virtio
            Name:   containerdisk
            Disk:
              Bus:  virtio
            Name:   cloudinitdisk
          Interfaces:
            Masquerade:
            Name:  default
        Machine:
          Type:  q35
        Resources:
          Requests:
            Memory:  64M
      Networks:
        Name:  default
        Pod:
      Volumes:
        Container Disk:
          Image:  docker-repo.nibr.novartis.net/kubevirt/cirros-container-disk-demo
        Name:     containerdisk
        Cloud Init No Cloud:
          userDataBase64:  SGkuXG4=
        Name:              cloudinitdisk
Status:
  Conditions:
    Last Probe Time:       2024-02-14T19:42:02Z
    Last Transition Time:  2024-02-14T19:42:02Z
    Message:               Guest VM is not reported as running
    Reason:                GuestNotRunning
    Status:                False
    Type:                  Ready
    Last Probe Time:       <nil>
    Last Transition Time:  <nil>
    Status:                True
    Type:                  LiveMigratable
    Last Probe Time:       <nil>
    Last Transition Time:  2024-02-14T19:40:45Z
    Message:               failed to configure vmi network: setup failed, err: open /proc/sys/net/ipv6/conf/all/forwarding: no such file or directory
    Reason:                Synchronizing with the Domain failed.
    Status:                False
    Type:                  Synchronized
  Created:                 true
  Desired Generation:      13
  Observed Generation:     13
  Printable Status:        Starting
  Volume Snapshot Statuses:
    Enabled:  false
    Name:     containerdisk
    Reason:   Snapshot is not supported for this volumeSource type [containerdisk]
    Enabled:  false
    Name:     cloudinitdisk
    Reason:   Snapshot is not supported for this volumeSource type [cloudinitdisk]
Events:
  Type    Reason            Age                    From                       Message
  ----    ------            ----                   ----                       -------
  Normal  SuccessfulDelete  62m                    virtualmachine-controller  Stopped the virtual machine by deleting the virtual machine instance 17d16d5c-9c94-4f2a-924f-2ee97fa23804
  Normal  SuccessfulDelete  52m                    virtualmachine-controller  Stopped the virtual machine by deleting the virtual machine instance 4e35ebe0-0e83-41f1-aa88-0e358506884f
  Normal  SuccessfulDelete  40m                    virtualmachine-controller  Stopped the virtual machine by deleting the virtual machine instance b7be5e98-af2b-4864-8561-67b8bfbc25c8
  Normal  SuccessfulDelete  29m                    virtualmachine-controller  Stopped the virtual machine by deleting the virtual machine instance 38b6357a-878a-48ce-adb0-03e963a351d8
  Normal  SuccessfulDelete  18m                    virtualmachine-controller  Stopped the virtual machine by deleting the virtual machine instance c2fadcd2-c165-4692-96a9-ec435f8d4ad4
  Normal  SuccessfulDelete  11m                    virtualmachine-controller  Stopped the virtual machine by deleting the virtual machine instance f37fac6e-09b2-4e72-b566-04b279bdd2b0
  Normal  SuccessfulCreate  6m34s (x240 over 41h)  virtualmachine-controller  Started the virtual machine by creating the new virtual machine instance testvm`

**Comment by EdDev** [comment]
Single stack IPv4 is fixed in v1.2.
It should also be available on a future v1.1.2 release if/when it will be available.

**Comment by seumassoltysik** [comment]
Terrific. Thank you folks.
